ament_copyright/ament_copyright/__init__.py

```diff
@@ -12,31 +12,354 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from __future__ import print_function
+
+import argparse
+import os
+import pkg_resources
+import re
+import sys
+import time
 from xml.sax.saxutils import escape
 from xml.sax.saxutils import quoteattr
 
+AMENT_COPYRIGHT_GROUP = 'ament_copyright.names'
+
+
+def main(argv=sys.argv[1:]):
+    extensions = [
+        'c', 'cc', 'cpp', 'cxx', 'h', 'hh', 'hpp', 'hxx',
+        'cmake',
+        'py',
+    ]
+
+    parser = argparse.ArgumentParser(
+        description='Check code for existance of a copyright reference.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        'paths',
+        nargs='*',
+        default=[os.curdir],
+        help="The files or directories to check. For directories files ending "
+             'in %s will be considered.' %
+             ', '.join(["'.%s'" % e for e in extensions]))
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        '--add-missing',
+        help='Add missing copyright notice with the passed copyright holder')
+    group.add_argument(
+        '--add-copyright-year',
+        action='store_true',
+        help='Add the current year to existing copyright notices')
+    group.add_argument(
+        '--list-names',
+        action='store_true',
+        help='List name of possible copyright holders')
+    # not using a file handle directly
+    # in order to prevent leaving an empty file when something fails early
+    group.add_argument(
+        '--xunit-file',
+        help='Generate a xunit compliant XML file')
+    args = parser.parse_args(argv)
+
+    names = get_copyright_names()
+    if args.list_names:
+        for key in sorted(names.keys()):
+            print('%s: %s' % (key, names[key]))
+        return 0
+
+    if args.xunit_file:
+        start_time = time.time()
+
+    filenames = get_files(args.paths, extensions)
+    if not filenames:
+        print('No files found', file=sys.stderr)
```

Comment:
Should this be prefixed by some context, e.g. `ament_copyright: No files found`? It might be confusion if you imagine this is the only output mixed in with other output from the other linters or from CMake.